internal/provider/resource_apply_module.go

```diff
@@ -0,0 +1,573 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/smithy-go/ptr"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/martian-cloud/terraform-provider-tharsis/internal/modifiers"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	sdktypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+type doRunInput struct {
+	model     *ApplyModuleModel
+	doDestroy bool
+}
+
+type doRunOutput struct {
+	moduleVersion string
+}
+
+// appliedModuleInfo contains what information was available about the latest applied run.
+// One or both fields may be nil, in which case information was not available.
+type appliedModuleInfo struct {
+	moduleSource  *string
+	moduleVersion *string
+}
+
+const (
+	jobCompletionPollInterval = 5 * time.Second
+)
+
+var (
+	applyRunComment = "terraform-provider-tharsis" // must be var, not const, to take address
+)
```

Comment:
Since this only contains one variable, it should be:
```suggestion
var applyRunComment = "terraform-provider-tharsis" // must be var, not const, to take address
```